source/dynamical_systems/src/DynamicalSystem.cpp

```diff
@@ -19,6 +20,17 @@ template<>
 DynamicalSystem<CartesianState>::DynamicalSystem(const std::string& base_frame) :
     base_frame_(CartesianState::Identity(base_frame, base_frame)) {}
 
+template<class S>
+bool DynamicalSystem<S>::is_compatible(const S&) {
+  throw exceptions::NotImplementedException("is_compatible(state) not implemented for this type of state");
```

Comment:
I think there could be a check as well for `JointState` but I was only interested in the `CartesianState` one for now